teps/0056-pipelines-in-pipelines.md

```diff
@@ -485,7 +495,239 @@ These are alternatives to the solution for status discussed above:
+    - name: notification
+      taskRef:
+        name: notification
+```
+
+#### Producing Results
+
+`Pipelines` in `Pipelines` will produce `Results`, in the same way as `Tasks` in `Pipelines`.
+
+`Results` produced by `TaskRuns` in a child `PipelineRun` that need to be passed to subsequent `PipelineTasks` will need
+to be propagated to the `Results` of the child `PipelineRun`.
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: clone-scan-notify
+spec:
+  params:
+    - name: repo
+      value: $(params.repo)
+  tasks:
+    - name: git-clone
+      taskRef:
+        name: git-clone
+    - name: security-scans
+      pipelineRef:
+        name: security-scans
+    - name: notification
+      params:
+        - name: commit
+          value: $(tasks.security-scans.results.scan-outputs) # --> result produced from pipelines in pipelines
+      taskRef:
+        name: notification
+```
+
+### Execution Status
+
+`Pipelines` in `Pipelines` will produce execution status that is consumed in `Finally Tasks`, in the same way as `Tasks`
+in `Pipelines`.
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: clone-scan-notify
+spec:
+  params:
+    - name: repo
+      value: $(params.repo)
+  tasks:
+    - name: git-clone
+      taskRef:
+        name: git-clone
+    - name: security-scans
+      pipelineRef:
+        name: security-scans
+  finally:        
+    - name: notification
+      params:
+        - name: commit
+          value: $(tasks.security-scans.status) # --> execution status produced from pipelines in pipelines
+      taskRef:
+        name: notification
+```
+
+### Workspaces 
+
+`PipelineTasks` with `Pipelines` can reference `Workspaces`, in the same way as `PipelineTasks` with `Tasks`. In these
+case, the `Workspaces` from the parent `PipelineRun` will be bound to the child `PipelineRun`. 
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: clone-scan-notify
+spec:
+  workspaces:
+    - name: output
+  tasks:
+    - name: git-clone
+      workspaces:
+        - name: output
+      taskRef:
+        name: git-clone
+    - name: security-scans
+      workspaces:
+        - name: output
+      pipelineRef:
+        name: security-scans
+    - name: notification
+      taskRef:
+        name: notification
+```
+
+### When Expressions
+
+Users can specify criteria to guard the execution of `PipelineTasks` using `when` expressions. When `when` expressions
+in a `PipelineTask` with a `Task` or `Custom Task` evaluate to `false`, the `PipelineTask` will be skipped - `TaskRun`
+or `Run` is not executed. In the same way, `PipelineTasks` with `Pipelines` will be skipped - `PipelineRun` will not be
+executed.
+
+```yaml
+  tasks:
+    - name: security-scans # --> skipped task 
+      when:
+        - input: foo
+          operator: in
+          values:
+            - bar
+      pipelineRef:
+        name: security-scans
+    ...
+```
+
+### Retries
```

Comment:
currently we have retries status for taskruns stored in their status. Is the proposal to update the pipelinerun status to add retries status info as well?